zerver/lib/events.py

```diff
@@ -1321,17 +1319,32 @@ def _draft_update_action(i: int) -> None:
                     if stream["stream_id"] in deleted_stream_ids:
                         stream["is_archived"] = True
                         if stream["first_message_id"] is None:
-                            stream["first_message_id"] = Stream.objects.get(
+                            new_first_message_id = Stream.objects.get(
                                 id=stream["stream_id"]
                             ).first_message_id
+                            assert new_first_message_id is not None
+                            stream["first_message_id"] = new_first_message_id
+                            updated_first_message_ids[stream["stream_id"]] = new_first_message_id
```

Comment:
I'm not sure I understand why archiving a channel might need to update `first_message_id` on it?

Reply:
Because we send a message when channel is archived and I guess a user who is not subscribed to the stream does not receive the message event.

Reply:
Ahh, OK.